### PR TITLE
chore(fastify): setup

### PR DESCRIPTION
**Pull request by JacopoDaeli**
Initial fastify setup project. This PR is open again the new branch `7x` (and not master).

### CODE REVIEW

**Comment by JacopoDaeli** [comment]
In short, warehouse as fastify plugin.

**Comment by JacopoDaeli** [comment]
@decompil3d @jcrugzz thoughts?

File: package.json [hunk 1]
```diff
@@ -2,10 +2,40 @@
...
+  },
+  "devDependencies": {
+    "eslint": "^7.10.0",
+    "eslint-config-godaddy": "^4.0.0",
+    "eslint-plugin-json": "^2.1.2",
+    "eslint-plugin-mocha": "^8.0.0",
+    "fastify-cli": "^2.3.0",
+    "nodemon": "^2.0.4",
+    "standard-version": "^9.0.0",
+    "tap": "^14.0.0"
```
Comment (decompil3d) [inline package.json:25]: Why not `mocha`?

File: lib/app.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  const fastify = createFastify({ logger: log });
+
+  fastify.decorate('verifyAuthentication', (request, reply, done) => {
+    done();
+  });
+
+  fastify.register(warehouse);
+
+  // eslint-disable-next-line no-process-env
+  fastify.listen(process.env.PORT, err => {
```
[the remark this answers is not in the document]
Comment (JacopoDaeli) [reply lib/app.js:16]: Yes. I was planning to add a config file with a separate PR, but let me add this directly within this PR.

File: package.json (hunk 1 above)
Comment (JacopoDaeli) [reply package.json:25]: `tap` is the preferred testing framework for `fastify` applications: https://www.fastify.io/docs/latest/Testing/

However, I haven't really thought about testing yet. I left this devDep from the default projects that gets generated with `npm init fastify`.

File: lib/app.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+const createFastify = require('fastify');
+const pino = require('pino');
```
[the remark this answers is not in the document]
Comment (JacopoDaeli) [reply lib/app.js:2]: Fastify default logger is `pino`. I could have indeed just wrote `createFastify({ logger: { level: 'info' }})` and that would have used created a `pino` instance underline. 
The reason why I required `pino` explicitly is to make it more clear and lock in the pino version.

File: package.json (hunk 1 above)
Comment (JacopoDaeli) [reply package.json:23]: Before writing the first test we can look into `mocha` vs `tap` deeper and choose the one we prefer.

**Review by jkeslin-godaddy — APPROVED** [review]

**Review by decompil3d — APPROVED** [review]

**Pull request merged**